Subject: rt: fix RCU menuconfig
From: Ingo Molnar <mingo@elte.hu>
Date: Mon Feb 09 22:32:04 CET 2009

PREEMPT_RT relies on PREEMPT_RCU - only allow RCU to be configured
interactively in the !PREEMPT_RT case.

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 init/Kconfig |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-2.6-tip/init/Kconfig
===================================================================
--- linux-2.6-tip.orig/init/Kconfig
+++ linux-2.6-tip/init/Kconfig
@@ -306,6 +306,7 @@ choice
 
 config CLASSIC_RCU
 	bool "Classic RCU"
+	depends on !PREEMPT_RT
 	help
 	  This option selects the classic RCU implementation that is
 	  designed for best read-side performance on non-realtime
@@ -315,6 +316,7 @@ config CLASSIC_RCU
 
 config TREE_RCU
 	bool "Tree-based hierarchical RCU"
+	depends on !PREEMPT_RT
 	help
 	  This option selects the RCU implementation that is
 	  designed for very large SMP system with hundreds or
